app/src/main/java/org/simple/clinic/summary/PatientSummaryUpdate.kt

```diff
@@ -179,7 +181,20 @@ class PatientSummaryUpdate(
       event: StatinInfoLoaded,
       model: PatientSummaryModel
   ): Next<PatientSummaryModel, PatientSummaryEffect> {
-    return next(model.updateStatinInfo(event.statinInfo))
+    val canShowSmokingStatusDialog =
+        event.statinInfo.canPrescribeStatin &&
+        (event.statinInfo.cvdRisk?.level == CVDRiskLevel.LOW_HIGH ||
+        event.statinInfo.cvdRisk?.level == CVDRiskLevel.MEDIUM_HIGH) &&
+        event.statinInfo.isSmoker == MedicalHistoryAnswer.Unanswered &&
+        !model.hasShownSmokingStatusDialog
```

Comment:
While this works, and we can leave this for now. The underlying problem still exists. The statin info is being reloaded every time patient info changes. So, the `LoadStatinPrescriptionCheckInfo` effect gets triggered every time.

Reply:
Thanks for flagging this out. I will create a story and move to backlog.